### DISCUSSION TITLE
package.json postbuild not automatically running after turbo run build

### DISCUSSION THREAD

**Discussion by cill-i-am** [vercel/turborepo#1027] [Help] [answered]
I'm trying to generate a sitemap.xml using `next-sitemap` but the postbuild script isn't running for me. If i run the script manually it works as expected, but `pnpm turbo run build` doesn't trigger it. I feel like i'm missing a step somewhere

`package.json`
```
 "scripts": {
    "build": "turbo run build",
    "dev": "turbo run dev --parallel",
    "lint": "turbo run lint",
    "test": "turbo run test",
    "format": "prettier --write \"**/*.{ts,tsx,md}\"",
    "prepare": "husky install",
    "postbuild": "turbo run postbuild"
  },
```

`apps/next-app`
```
"scripts": {
    "dev": "next dev --port=3000",
    "build": "next build",
    "start": "next start",
    "lint": "next lint",
    "test": "jest --coverage",
    "test-watch": "jest --watch",
    "postbuild": "next-sitemap --config next-sitemap.config.js"
  },
```

**Comment by Barik85** [comment]
Any suggestions for this issue?

**Reply by Barik85** [reply]
I tried to add postbuild script manually `yarn prebuild && yarn turbo run build --scope=my-app --include-dependencies --no-deps && yarn postbuild`
in `turbo.json`:
```
 "postbuild": {
      "dependsOn": ["^build"],
      "outputs": ["dist/**", ".next/**"]
    },
```
It look like it works only once. Because on my first build i got robots.txt on preview domain. But next times i got 404.

**Comment by samuelcole** [comment]
i'm having the same problem... my theory is that the generated files in `public/` aren't part of the turbo build cache

**Comment by samuelcole** [comment, marked as answer]
fyi, for anyone who comes across this, i solved it by breaking `sitemap` into it's own task:

```
sitemap": {
  "dependsOn": ["build"],
  "outputs": ["public/robots.txt", "public/sitemap*.xml"]
},
```

and then a `deploy` task which both builds and sitemaps:

```
"deploy": {
  "dependsOn": ["build", "sitemap"],
  "outputs": []
},
```

**Reply by andhisan** [reply]
> fyi, for anyone who comes across this, i solved it by breaking `sitemap` into it's own task:
> 
> ```
> sitemap": {
>   "dependsOn": ["build"],
>   "outputs": ["public/robots.txt", "public/sitemap*.xml"]
> },
> ```
> 
> and then a `deploy` task which both builds and sitemaps:
> 
> ```
> "deploy": {
>   "dependsOn": ["build", "sitemap"],
>   "outputs": []
> },
> ```

Also note that we need to change name of `postbuild` script to `sitemap`.

**Reply by jaredcbaum** [reply]
For extra clarification, you will also have to change your build command in your app deployment settings. I'm using Vercel, so now my build command looks like this: `cd ../.. && npx turbo run deploy --filter=[my-app-name]...`
